scripts/linters/pylint_extensions_test.py

```diff
@@ -3257,3 +3257,93 @@ def some_random_function(param1, param2):
 
         with self.checker_test_object.assertNoMessages():
             self.checker_test_object.checker.visit_functiondef(def_node)
+
+
+class ConcatenationCheckerTests(unittest.TestCase):
+
+    def setUp(self):
+        super(ConcatenationCheckerTests, self).setUp()
+        self.checker_test_object = testutils.CheckerTestCase()
+        self.checker_test_object.CHECKER_CLASS = (
+            pylint_extensions.StringConcatenationChecker)
+        self.checker_test_object.setup_method()
+
+    def test_string_concatenation_with_string_operands_raises_error(self):
+        binop_node1, binop_node2, binop_node3 = astroid.extract_node(
+            """
+        'string1' + 'string2' #@
+        a + 'string1' #@
+        'string2' + b #@
+        """)
+
+        str_str_message = testutils.Message(
+            msg_id='string-concatenation', args=('\'string1\''),
+            node=binop_node1)
+        right_message = testutils.Message(
+            msg_id='string-concatenation', args=('\'string1\''),
+            node=binop_node2)
+        left_message = testutils.Message(
+            msg_id='string-concatenation', args=('\'string2\''),
+            node=binop_node3)
+
+        with self.checker_test_object.assertAddsMessages(
+            str_str_message, right_message, left_message):
+            self.checker_test_object.checker.visit_binop(binop_node1)
+            self.checker_test_object.checker.visit_binop(binop_node2)
+            self.checker_test_object.checker.visit_binop(binop_node3)
+
+    def test_string_interpolation_does_not_raise_error(self):
+        binop_node1, binop_node2 = astroid.extract_node(
+            """
+        'string1%s' % (string2) #@
+        '%sstring2' % d #@
+        """)
+
+        with self.checker_test_object.assertNoMessages():
+            self.checker_test_object.checker.visit_binop(binop_node1)
+            self.checker_test_object.checker.visit_binop(binop_node2)
+
+    def test_addition_of_non_string_variables_does_not_raise_error(self):
```

Comment:
Have you tested the case when we concatenate 2 string variables?

Reply:
Done !!!!!!!!